actors/builtin/miner/miner_test.go

```diff
@@ -2616,6 +2679,253 @@ func TestChangeMultiAddrs(t *testing.T) {
+		require.EqualValues(t, secondExpiation, info.PendingBeneficiaryTerm.NewExpiration)
+
+		//confirm by old beneficiary address
+		actor.changeBeneficiary(rt, firstBeneficiary, beneficiaryChange{secondBeneficiaryId, secondQuota, secondExpiation}, nil)
+		info = actor.getInfo(rt)
+		require.EqualValues(t, secondBeneficiaryId, info.PendingBeneficiaryTerm.NewBeneficiary)
+		require.EqualValues(t, secondQuota, info.PendingBeneficiaryTerm.NewQuota)
+		require.EqualValues(t, secondExpiation, info.PendingBeneficiaryTerm.NewExpiration)
+
+		//confirm by new beneficiary address and worked
+		actor.changeBeneficiary(rt, secondBeneficiary, beneficiaryChange{secondBeneficiaryId, secondQuota, secondExpiation}, &secondBeneficiaryId)
+		info = actor.getInfo(rt)
+		require.EqualValues(t, secondBeneficiaryId, info.Beneficiary)
+		require.EqualValues(t, secondQuota, info.BeneficiaryTerm.Quota)
+		require.EqualValues(t, secondExpiation, info.BeneficiaryTerm.Expiration)
+		require.EqualValues(t, big.Zero(), info.BeneficiaryTerm.UsedQuota)
+		require.EqualValues(t, (*miner.PendingBeneficiaryChange)(nil), info.PendingBeneficiaryTerm)
+	})
+
+	t.Run("failure approval has different params", func(t *testing.T) {
+		rt, actor := setupFunc()
+		actor.constructAndVerify(rt)
+
+		quota := abi.NewTokenAmount(100)
+		expiration := periodOffset * 2
+
+		rt.ExpectAbort(exitcode.ErrForbidden, func() {
+			actor.changeBeneficiary(rt, firstBeneficiary, beneficiaryChange{firstBeneficiaryId, quota, expiration + 1}, &firstBeneficiary)
+		})
+
+		actor.changeBeneficiary(rt, actor.owner, beneficiaryChange{firstBeneficiaryId, quota, expiration}, nil)
+		// assert change has been made in state
+		info := actor.getInfo(rt)
+		require.EqualValues(t, expiration, info.PendingBeneficiaryTerm.NewExpiration)
+		require.EqualValues(t, quota, info.PendingBeneficiaryTerm.NewQuota)
+		require.EqualValues(t, firstBeneficiaryId, info.PendingBeneficiaryTerm.NewBeneficiary)
+
+		rt.SetEpoch(periodOffset + 10)
+		rt.ExpectAbort(exitcode.ErrIllegalArgument, func() {
+			actor.changeBeneficiary(rt, firstBeneficiary, beneficiaryChange{firstBeneficiaryId, quota, expiration + 1}, &firstBeneficiary)
+		})
+
+		rt.ExpectAbort(exitcode.ErrIllegalArgument, func() {
+			actor.changeBeneficiary(rt, firstBeneficiary, beneficiaryChange{firstBeneficiaryId, big.Add(quota, big.NewInt(1)), expiration}, &firstBeneficiary)
+		})
+
+		rt.ExpectAbort(exitcode.ErrIllegalArgument, func() {
+			actor.changeBeneficiary(rt, firstBeneficiary, beneficiaryChange{secondBeneficiaryId, big.Add(quota, big.NewInt(1)), expiration}, &firstBeneficiary)
+		})
+	})
+
+	t.Run("failure change beneficiary has wrong params", func(t *testing.T) {
+		rt, actor := setupFunc()
+		actor.constructAndVerify(rt)
+
+		rt.SetEpoch(periodOffset)
+		rt.ExpectAbort(exitcode.ErrIllegalArgument, func() {
+			actor.changeBeneficiary(rt, actor.owner, beneficiaryChange{firstBeneficiary, abi.NewTokenAmount(10), periodOffset - 10}, nil)
```

Comment:
Should this be using `firstBeneficiaryId`? I'm not quite sure. Could you document which check each of these is verifying?

Reply:
both id/address are okay, but here have a dup test item,  would remove one. and  add one test for beneficiary address

Reply:
comment for each case